booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,645 @@
+  );
+
+  private static final JsonContentCheck VALIDATE_SHIPMENT_LOCATIONS = JsonAttribute.customValidator(
+    "Validate shipmentLocations",
+    body -> {
+      var issues = new LinkedHashSet<String>();
+      var shipmentLocations = body.path("shipmentLocations");
+      var receiptTypeAtOrigin = body.path("receiptTypeAtOrigin").asText("");
+      var polNode =
+        StreamSupport.stream(shipmentLocations.spliterator(), false)
+          .filter(o ->  o.path("locationTypeCode").asText("").equals("POL")
+            || o.path("locationTypeCode").asText("").equals("PRE") )
+          .findFirst()
+          .orElse(null);
+      var podNode =
+        StreamSupport.stream(shipmentLocations.spliterator(), false)
+          .filter(o -> o.path("locationTypeCode").asText("").equals("POD")
+          || o.path("locationTypeCode").asText("").equals("PDE") )
+          .findFirst()
+          .orElse(null);
+
+      var ielNode =
+        StreamSupport.stream(shipmentLocations.spliterator(), false)
+          .filter(o ->  o.path("locationTypeCode").asText("").equals("IEL"))
+          .findFirst()
+          .orElse(null);
+
+      if (polNode == null || polNode.isEmpty() ) {
+        issues.add("Port of Discharge/Place of Delivery value must be provided");
+      }
+      if (podNode == null || podNode.isEmpty()) {
+        issues.add("Port of Load/Place of Receipt values must be provided");
+      }
+      if(!"SD".equals(receiptTypeAtOrigin) && ielNode != null) {
+        issues.add("Container intermediate export stop-off location should not be provided");
+      }
+
+      return issues;
+    });
+
+
+  private static final JsonContentCheck REQUESTED_CHANGES_PRESENCE = JsonAttribute.customValidator(
+    "Requested changes must be present for the selected Booking Status ",
+    (body) -> {
+      var bookingStatus = body.path("bookingStatus").asText("");
+      var issues = new LinkedHashSet<String>();
+      if (PENDING_CHANGES_STATES.contains(BookingState.fromWireName(bookingStatus))) {
+        var requestedChanges = body.get("requestedChanges");
+        if (requestedChanges == null) {
+          issues.add("requestedChanges is missing in allowed booking states %s".formatted(PENDING_CHANGES_STATES));
+        }
+      }
+      return issues;
+    }
+  );
+
+  private static final JsonContentCheck REASON_PRESENCE = JsonAttribute.customValidator(
+    "Reason field must be present for the selected Booking Status",
+    (body) -> {
+      var bookingStatus = body.path("bookingStatus");
+      var amendedBookingStatus = body.path("amendedBookingStatus");
+      var issues = new LinkedHashSet<String>();
+      var status = amendedBookingStatus.isMissingNode() || amendedBookingStatus.isNull() ? bookingStatus : amendedBookingStatus;
+      if (REASON_STATES.contains(BookingState.fromWireName(status.asText()))) {
+        var reason = body.get("reason");
+        if (reason == null) {
```

Comment:
I am sure it was confirmed (somewhere on Teams) that it is not a requirement to add a `reason` - in some cases it is optional

Reply:
I am checking only the states where its must to have.

Reply:
Sorry - I misunderstood this check - have you made a check somewhere else that checks that a `reason` field should not be provided in a Confirmed Booking?